src/draft-blockchain.js

```diff
@@ -0,0 +1,136 @@
+const UTXO = require('UTXO')
+const Block = require('block')
+const mining_target = require('CONFIG')
+const Transaction = require('transaction')
+const genesis_coinbase = require('genesis')
+
+let current_blockchain
+
+const get_blockchain = () => {
+  if (current_blockchain === undefined) {
+    current_blockchain = new Blockchain()
+  }
+  return current_blockchain
+}
+
+class Blockchain{
+  constructor(){
+    this.chain = [genesis_coinbase]
+  }
+
+
+  insertBlock(block) {
+    if (!(block instanceof Block)) {
+      return false;
+    }   
+    for (let tx of block.transactions) {
+      if (!tx.isValid()) {
+        return false;
+      }
+      if (tx instanceof Transaction) {
+        for (let utxo of tx.utxos) {
+          if (!this.isValidUTXO(utxo)) {
+            return false;
+          }
+        }
+      }
+    }
+    if (!this.checkAgainstTarget(mining_target, block.getHash())) {
+      return false;
+    }
+    this.chain.push(block);
+    return true;
+  }
+
+
+  isChainValid(){
+    for(let i = 1; i < this.chain.length; i++){
```

Comment:
`this.chain.length` = 1 before you insert into a block. This will result to undefined returned by the method itself

Reply:
thinking about setting it this way:

isChainValid() {
  if (this.chain.length === 1) {
    throw new Error("Chain only contains the genesis block");
    ....."function code"
  }

Reply:
yeah, makes sense to throw an error it contains only genesis block 👍